Add support for the inbound-cidrs annotation

MarkFreebairn opened the pull request:
If the `alb.ingress.kubernetes.io/inbound-cidrs` annotation is set then the more restrictive CIDR IPv4 range(s) will be added (vs. the default `0.0.0.0/0`).

Fix #294.

I paired with @erran on these changes.

# Test cases
The following test cases were considered/run manually:

- [x] Starting with no annotation (inbound cidr defaults to 0.0.0.0/0 on managed sg) -> adding annotation adds new inbound cidr range and removes default 0.0.0.0/0
- [x] appending new cidr to annotation will add the new inbound source to the managed sg
- [x] removing a cidr from the annotation will remove it from the sg
- [x] removing the annotation entirely will remove all inbound cidrs and add the default 0.0.0.0/0
- [x] adding a new port will add all inbound cidrs from the annotation

MarkFreebairn commented:
Hey @joshrosso would you mind having a look at this PR if you have some spare time please?

bigkraig commented on pkg/alb/loadbalancer/loadbalancer.go:
```diff
@@ -547,6 +555,14 @@ func (lb *LoadBalancer) needsModification() (loadBalancerChange, bool) {
 	}
 
 	if lb.CurrentPorts != nil && lb.CurrentManagedSG != nil {
+		if lb.CurrentInboundCidrs != nil {
```
Not 100% sure if this is required. If CurrentInboundCidrs is nil but there is a desired inbound cidr, it won't return with true.

bigkraig commented on pkg/annotations/annotations.go:
```diff
@@ -33,6 +33,7 @@ const (
 	healthcheckTimeoutSecondsKey  = "alb.ingress.kubernetes.io/healthcheck-timeout-seconds"
 	healthyThresholdCountKey      = "alb.ingress.kubernetes.io/healthy-threshold-count"
 	unhealthyThresholdCountKey    = "alb.ingress.kubernetes.io/unhealthy-threshold-count"
+	inboundCidrsKey               = "alb.ingress.kubernetes.io/inbound-cidrs"
```
If `inbound-cidrs` is a configuration of the security group, should this be something like `security-group-inbound-cidrs` ?

bigkraig commented on pkg/aws/ec2/ec2.go:
```diff
@@ -264,24 +285,36 @@ func (e *EC2) UpdateSGIfNeeded(vpcID *string, sgName *string, currentPorts []int
 	// for each addPort, run an authorize to ensure it's added
 	for _, port := range desiredPorts {
-		if existsInOtherPortRange(port, currentPorts) {
+		ipranges := []*ec2.IpRange{}
```
should this be ipRanges?

bigkraig commented on pkg/aws/ec2/ec2.go:
```diff
@@ -293,22 +326,29 @@ func (e *EC2) UpdateSGIfNeeded(vpcID *string, sgName *string, currentPorts []int
 	// for each currentPort, run a revoke to ensure it can be removed
 	for _, port := range currentPorts {
-		if existsInOtherPortRange(port, desiredPorts) {
+		ipranges := []*ec2.IpRange{}
```
ipRanges

bigkraig commented on pkg/aws/ec2/ec2.go:
```diff
@@ -371,18 +420,24 @@ func (e *EC2) CreateSecurityGroupFromPorts(vpcID *string, sgName *string, ports
+
 	// for every port specified, allow all tcp traffic.
 	for _, port := range ports {
+		ipranges := []*ec2.IpRange{}
```
ipRanges

MarkFreebairn commented:
Hey @bigkraig thanks a lot for reviewing, sorry it took me so long to get back to looking at this! I have updated based on your comments, fixed merge conflicts and ran through all of the manual tests as well to make sure everything is still working correctly.

bigkraig commented on pkg/alb/loadbalancer/loadbalancer.go:
```diff
@@ -586,14 +586,12 @@ func (lb *LoadBalancer) needsModification() (loadBalancerChange, bool) {
+			changes |= managedSecurityGroupsModified
 		}
-
+		
```
@MarkFreebairn sorry it took me until now for this minor request. apparently i had left it in pending for the last month

MarkFreebairn replied:
@bigkraig ah whoops sorry missed this! I've updated to fix merge conflicts and remove the additional whitespace now
